### PR TITLE
Read nodesCIDR directly from infrastructure status.

### PR DESCRIPTION
**Pull request by Gerrit91**
This is a workaround for accelerating the time for shoot provisioning. Should be able to be reverted at some point in the future.

### CODE REVIEW

File: pkg/controller/controlplane/valuesprovider.go [hunk 1]
```diff
@@ -667,14 +686,3 @@ func getLimitValidationWebhookControlPlaneChartValues(cluster *extensionscontrol
 
 	return values, nil
 }
-
-// Data for configuration of IDM-API WebHook (deployment to be done!)
-type UserDirectory struct {
```
Comment (Gerrit91) [inline pkg/controller/controlplane/valuesprovider.go:672]: Unrelated but can be removed as discussed with @droid42

File: pkg/controller/controlplane/valuesprovider.go [hunk 2]
```diff
@@ -370,8 +379,17 @@ func (vp *valuesProvider) GetControlPlaneChartValues(
 		return nil, err
 	}
 
+	// TODO: this is a workaround to speed things for the time being...
+	// the infrastructure controller writes the nodes cidr back into the infrastructure status, but the cluster resource does not contain it immediately
+	// it would need the start of another reconcilation until the node cidr can be picked up from the cluster resource
+	// therefore, we read it directly from the infrastructure status
+	infrastructure, err := vp.gardenerExtensionsClientset.ExtensionsV1alpha1().Infrastructures(cluster.Seed.Namespace).Get(cluster.Shoot.Name, metav1.GetOptions{})
```
Comment (rfranzke) [inline pkg/controller/controlplane/valuesprovider.go:386]: Why don't you use the client that you already have?

```suggestion
  infrastructure := &extensionsv1alpha1.Infrastructure{}
  if err := vp.client.Get(ctx, kutil.Key(cp.Namespace, cp.Name); infrastructure); err != nil {
    return nil, err
  }
```

Comment (Gerrit91) [reply pkg/controller/controlplane/valuesprovider.go]: Oh... I thought with this one you could only get Kubernetes objects. That looks of course much better. Thank you.

File: pkg/controller/controlplane/valuesprovider.go (hunk 1 above)
Comment (majst01) [reply pkg/controller/controlplane/valuesprovider.go:672]: can you please create a issue of this topic to be sure not to miss this.

**Pull request merged**